src/common/platform/posix/sdl/i_system.cpp

```diff
@@ -129,27 +134,57 @@ void CalculateCPUSpeed()
 
 void I_PrintStr(const char *cp)
 {
-	// Strip out any color escape sequences before writing to debug output
-	TArray<char> copy(strlen(cp) + 1, true);
 	const char * srcp = cp;
-	char * dstp = copy.Data();
+	FString printData = "";
 
 	while (*srcp != 0)
 	{
-		if (*srcp != 0x1c && *srcp != 0x1d && *srcp != 0x1e && *srcp != 0x1f)
+		if (*srcp == 0x1c && con_printansi)
 		{
-			*dstp++ = *srcp++;
+			srcp += 1;
+			EColorRange range = V_ParseFontColor((const uint8_t*&)srcp, CR_UNTRANSLATED, CR_YELLOW);
+			if (range != CR_UNDEFINED)
+			{
+				PalEntry color = V_LogColorFromColorRange(range);
+				if (con_4bitansi)
+				{
+					float h, s, v, r, g, b;
+					int attrib = 0;
+
+					RGBtoHSV(color.r / 255.f, color.g / 255.f, color.b / 255.f, &h, &s, &v);
+					if (s != 0)
+					{ // color
+						HSVtoRGB(&r, &g, &b, h, 1, 1);
+						if (r == 1)  attrib  = 0x4;
+						if (g == 1)  attrib |= 0x2;
+						if (b == 1)  attrib |= 0x1;
+						if (v > 0.6) attrib |= 0x8;
+					}
+					else
+					{ // gray
+						     if (v < 0.33) attrib = 0x8;
+						else if (v < 0.90) attrib = 0x7;
+						else			   attrib = 0x15;
+					}
+					static unsigned char cga_to_ansi[] = { 0, 4, 2, 6, 1, 5, 3, 7 };
```

Comment:
Ansi simply reverses bits 0 and 2 (0x1 and 0x4 respectively), there is no need for a lookup table here.

Reply:
That should be easy as swapping 0x3,0x7 and 0x1,0x4.